ui/src/elle/tools/correction/Correction.component.js

```diff
@@ -1,592 +1,721 @@
+        fetch("/api/texts/keerukus", {
+            method: "POST", headers: {
+                'Accept': 'application/json', 'Content-Type': 'application/json'
+            }, body: JSON.stringify(obj)
+        }).then(v => v.json()).then(t => {
+            console.log(t)
+            setComplexityAnswer(t);
+        });
+    };
 
-  kysi5 = () => {
-    let obj = {};
-    obj["tekst"] = this.state.alasisu;
-    fetch("/api/texts/keerukus", {
-      method: "POST",
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify(obj)
-    }).then(v => v.json()).then(t => {
-      this.setState({"keerukusvastus": t});
-    })
-  }
+    const getLanguageDiversity = () => {
+        let obj = {};
+        obj["tekst"] = content;
+        fetch("/api/texts/mitmekesisus", {
+            method: "POST", headers: {
+                'Accept': 'application/json', 'Content-Type': 'application/json'
+            }, body: JSON.stringify(obj)
+        }).then(v => v.json()).then(t => {
+            setDiversityAnswer(t);
+        });
+    };
 
-  kysi6 = () => {
-    let obj = {};
-    obj["tekst"] = this.state.alasisu;
-    fetch("/api/texts/mitmekesisus", {
-      method: "POST",
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify(obj)
-    }).then(v => v.json()).then(t => {
-      this.setState({"mitmekesisusvastus": t});
-    })
-  }
+    const handleTabChange = (event, newValue) => {
+        setOpenCard(newValue);
+    };
 
+    const getCorrections = async () => {
+        if (content === correctorAnswer[1]) {
+            return;
+        }
+        const obj = {
+            tekst: content
+        };
+        try {
+            const response = await fetch("/api/texts/korrektuur", {
+                method: "POST", headers: {
+                    'Accept': 'application/json', 'Content-Type': 'application/json'
+                }, body: JSON.stringify(obj)
+            });
+
+            if (!response.ok) {
+                throw new Error('Request failed');
+            }
+
+            const data = await response.json();
+            let sm = data[1].split(" ");
+            let vm = data[0].split(" ");
+            await setContentWords(sm);
+            await setResponseWords(vm);
+            await fillData(sm, vm);
+        } catch (error) {
+            throw new Error('Request failed');
+        }
+    };
 
-  korrektuuriVajutus = () => {
-    this.setState({avatudkaart: "korrektuur"})
-    if (this.state.kordab) {
-      this.kysi3();
+    const fillData = (content, answer) => {
+        const answerText = replyCode;
+        let contentText = "";
+        const changes = [];
+        const contentPlacement = [];
+        if (content === answer) {
+            setCorrection("Kõik korras");
+        } else {
+            for (let i = 0; i < answer.length; i++) {
```

Comment:
tavalised for loopid võiks olla ainult erandjuhtudel kasutuses, pigem teha foreachiga